### ISSUE TITLE
Resources with typed arrays of resources show up as the path to the resource instead of the class name in editor causing an error that the type does not match.

### ISSUE THREAD

**Issue by ryanbbernstein** [godotengine/godot#95851]
### Tested versions

- Reproducible in 4.3 but was not present in 4.2

### System information

Mac OS Sonoma M1

### Issue description

Resources with typed arrays of resources show up as the path to the resource instead of the class name in editor causing an error that the type does not match. Removing the typing in script logic accessing the array fixes the error and the script runs correctly.

Resource contains:
```gdscript
@export var resources: Array[test_res]
```
Accessing it like so:
```gdscript
const CONTAINER: container_res = preload("res://container.tres")
...
for resource: test_res in CONTAINER.resources:
```
Produces:
`Parse Error: Unable to iterate on value of type "Array[res://test_res.gd]" with variable of type "test_res".`

I believe this is related to the following changes: #78219 #90751 #85024

### Steps to reproduce

Create resource that contains an array of custom resources. Attempt to iterate or access values from the array as the custom resource class type.

### Minimal reproduction project (MRP)

Minimum example files
[testgd.zip](https://github.com/user-attachments/files/16671595/testgd.zip)

**Comment by matheusmdx** [comment]
This looks like a duplicate of #95568

**Comment by ryanbbernstein** [comment]
Yep, looks like it. I searched through the existing issue and couldn't find anything related. Probably because the title doesn't include the word resource. Thanks
